File: frontend/src/types/property.ts
export interface PropertyBase {
  title: string;
  description?: string | null;
  price: number;
  location: string;
  area?: number | null;
}

export type PropertyCreate = PropertyBase;
export type PropertyUpdate = Partial<PropertyBase>;

export interface Property extends PropertyBase {
  id: number;
  created_at?: string | null;
}

File: backend/app/schemas/property.py
# app/schemas/property.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime

class PropertyCreate(BaseModel):
    title: str
    description: Optional[str]
    price: float
    location: str
    area: Optional[float]

class PropertyUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    price: Optional[float] = None
    location: Optional[str] = None
    area: Optional[float] = None

class PropertyResponse(PropertyCreate):
    id: int
    created_at: Optional[datetime]

    class Config:
        orm_mode = True
